src/packages.rkt

```diff
@@ -292,18 +292,51 @@
 
 (define (package-search text tags)
   (define res (map (lambda (r) (regexp (regexp-quote r #f))) (string-split text)))
-  (define packages (manager-rpc 'packages))
-  (sort-package-names
-   (filter (lambda (package-name)
```

Comment:
Was there any reason to rewrite this code? It looks the same to me, only taller.

(This diff was much harder to read than the previous one)

Reply:
Alright. I didn't consider readability of the diff previously. I was mostly trying to make the code more consistent (operating on list of cons throughout the execution. Using the `filter` for both tagging stage and search text stage, etc.) and avoid unnecessarily `hash-ref`, but I agree that it probably is not worth touching it. Will fix this by basing on the old code, but changing `sort-package-names` to `sort-package-names/priority`.

Reply:
Oh, avoiding `hash-ref` in the sorting would be better.

I think it'd also be good to:
- put the sorting functions at the top level,
- and pre-process the text before any sorting happens,